askama_parser/src/expr.rs

```diff
@@ -927,6 +927,27 @@ impl<'a> Suffix<'a> {
             }
         }
 
+        fn lifetime<'a>(i: &mut InputStream<'a>) -> ParseResult<'a, ()> {
+            // Before the 2021 edition, we can have whitespace characters between "r#" and the
+            // identifier so we allow it here.
+            let start = *i;
+            '\''.parse_next(i)?;
+            if opt(alt((
+                (keyword("r#"), opt(skip_ws0), identifier).map(|(_, _, _)| ()),
```

Comment:
The identifier of a raw lifetime must not be `"crate" | "self" | "super" | "Self" | "_"`. For "normal" lifetimes the identifiers are okay, though. <https://doc.rust-lang.org/reference/tokens.html#grammar-LIFETIME_TOKEN>

Actually, raw lifetimes seem to be only useful as labels. A lifetime `'break` is okay, but `'break loop { break 'break; }` is not.